src/app/layout.tsx

```diff
@@ -0,0 +1,29 @@
+import type { Metadata } from 'next';
+import { Inter } from 'next/font/google';
+import './globals.css';
+import RecoilRootProvider from '@utils/RecoilRootProvider';
+
+const inter = Inter({ subsets: ['latin'] });
+
+export const metadata: Metadata = {
+  title: 'Songess-seungwan netlix clone by nextJS',
+  description:
+    'This website is netlix clone coding using nextJS and themovieDB API',
+};
+
+export default function RootLayout({
+  children,
+}: Readonly<{
+  children: React.ReactNode;
+}>) {
+  return (
+    <html lang="en">
+      <head>
+        <link rel="icon" href="/icon/netflix.ico" type="image/x-icon" />
+      </head>
+      <body className={inter.className}>
+        <RecoilRootProvider>{children}</RecoilRootProvider>
+      </body>
```

Comment:
코드를 보니 각 페이지별 layout.tsx 에 

```
<main className="flex-1 w-full overflow-y-scroll">{children}</main>;
```

이 부분이 많이 겹치는 것 같더라구여!

루트 layout.tsx 코드를

```suggestion
<RecoilRootProvider>
      <main className="flex-1 w-full overflow-y-scroll">
   {children}
      </main>
</RecoilRootProvider>
```

위처럼 설정해준다면 MainHeader를 가지고 있는 main 페이지의 세부 레이아웃만 따로 설정해주면 돼서 개발 cost가 줄어들 것 같습니다 ㅎㅎ